atomistics/calculators/wrapper.py

```diff
@@ -0,0 +1,71 @@
+"""
+A wrapper for mapping between functions that evaluate a single structure to those
+that evaluate a task dictionary.
+"""
+
+from __future__ import annotations
+
+from typing import Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ase import Atoms
+
+
+def _convert_task_dict(
+    old_task_dict: dict[str, dict[str, Atoms]]
+) -> dict[str, tuple[Atoms, list[str]]]:
+    """
+    Converts the existing task dictionaries of the format
+    `{result_type_string: {structure_label_string: structure, ...}, ...}`
+    to the new format
+    `{structure_label_string: (structure, [result_type_string, ...]), ...}`.
+
+    Can be removed if/when the rest of the codebase passing in these task
+    dictionaries gets updated to the new format.
+    """
+    task_dict = {}
+    for method_name, subdict in old_task_dict.items():
+        for label, structure in subdict.items():
+            try:
+                task_dict[label][1].append(method_name)
+            except KeyError:
+                task_dict[label] = (structure, [method_name])
+    return task_dict
+
+
+def task_evaluation(
+    calculate: callable[[Atoms, list[str], ...], dict[str, Any]],
+) -> callable[[dict[str, dict[str, Atoms]], ...], dict[str, dict[str, Any]]]:
```

Comment:
```suggestion
) -> Callable[[dict[str, dict[str, Atoms]], ...], dict[str, dict[str, Any]]]:
```

together with `from typing import Callable` I guess.

Reply:
Ah, this is 3.11 syntax.  On that point, I've had trouble getting mypy to recognize this, any hints @liamhuber ?

Reply:
Yep, definitely the sexy new syntax! I would prefer to leave it the new way unless absolutely necessary as I find it much cleaner.

No, I'm afraid I haven't played with mypy.